pkgs/servers/web-apps/funkwhale/update.sh

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env nix-shell
+#! nix-shell -i oil -p jq sd nix-prefetch-git ripgrep moreutils oil nixUnstable
+
+# TODO set to `verbose` or `extdebug` once implemented in oil
+shopt --set verbose
+
+const directory = $(dirname $0 | xargs realpath)
+const domain = "dev.funkwhale.audio"
+const owner = "funkwhale"
+const repo = "funkwhale"
+const latest_rev = $(curl -q https://${domain}/api/v4/projects/${owner}%2F${repo}/repository/tags | \
+  jq -r 'first.name')
+const latest_version = $(echo $latest_rev)
+const current_version = $(jq -r '.version' $directory/pin.json)
+if ($latest_version === $current_version) {
+  echo "funkwhale is already up-to-date"
+  return 0
+} else {
+  const source = "https://$domain/$owner/$repo/-/raw/$latest_version"
+  const package_json = $(curl -qf $source/front/package.json)
+  echo $package_json > $directory/package.json
+
+  const repo_meta = $(nix-prefetch-git https://${domain}/${owner}/${repo}.git --rev $latest_rev)
+  const repo_hash = $(echo $repo_meta | jq -r '.sha256')
+
+  jq ".version = \"$latest_version\" | \
+      .\"sha256\" = \"$repo_hash\" | \
+      .\"uiYarnDepsSha256\" = \"0000000000000000000000000000000000000000000000000000000000000000\"" $directory/pin.json | sponge $directory/pin.json
+
+  # position of line with hash is nix version dependant :(
+  const new_offline_cache_sha256 = $(nix-build -A funkwhale 2>&1 | \
+    tail -n 16 | \
+    head -n 1 | \
+    sd '\s+got:\s+' '')
```

Comment:
Use a `prefetch-yarn-deps` invocation instead of this unholy mess.

Reply:
Still looking for inspiration here.

Reply:
Am I correct that this greps for a hash mismatch in lack of proper prefetching? Maybe improve the regex so that at least we don't need to match exact line positions? (something something `got:.*\nexpected:.*`, but I hate regex)

Reply:
I want to use prefetch-yarn-deps if possible. Grepping the hash mismatch is a bad joke.

Reply:
Resolved.